styx-scheduler-service/src/main/java/com/spotify/styx/state/handlers/TerminationHandler.java

```diff
@@ -136,8 +136,9 @@ private boolean shouldFailFast(RunState state, Optional<Integer> exitCode) {
 
     var workflow = workflows.get().get(state.workflowInstance().workflowId());
     if (workflow == null) {
-      LOG.info("Workflow {} does not exist", state.workflowInstance().workflowId().toKey());
-      return true;
+      LOG.debug("Workflow {} does not exist possibly due to stale caching",
+          state.workflowInstance().workflowId().toKey());
+      return false;
```

Comment:
@NatashaL I had second thought of this. Not entirely impossible that we have stale cache when reading the workflow, so . here we just let it retry normally and it's up to `ExecutionDescriptionHandler` to decide later.